api/docker-entrypoint.sh

```diff
@@ -0,0 +1,177 @@
+#!/bin/bash
+
+info()
+{
+    echo "Info: $*"
+}
+
+warn()
+{
+    echo "Warning: $*"
+}
+
+fatal()
+{
+    echo "Fatal: $*"
+}
+
+get_couchdb_url()
+{
+  echo "http://$COUCHDB_USER:$COUCHDB_PASSWORD@$COUCHDB_SERVICE_NAME:5985"
+}
+
+wait_for_couchdb()
+{
+  local n=60
+
+  while [ "$n" -gt 0 ]; do
+    curl "`get_couchdb_url`" &>/dev/null
+    [ "$?" -eq 0 ] && return 0
+    sleep 3
+    n=$[$n-1]
+  done
+
+  return 1
+}
+
+is_setup_needed()
+{
+  ! is_existing_user 'medic-api'
+}
+
+is_existing_user()
+{
+  local user="$1"
+  local url="`get_couchdb_url`"
+  local id="org.couchdb.user:$user"
+  shift 1
+
+  local userdoc=$(curl -X GET $url/_users/$id | jq '._id?')
+
+  if [ "$userdoc" = \"org.couchdb.user:$user\" ]; then
+    if [ ! -f /srv/storage/$user/passwd/$user ]; then
+      create_couchdb_admin "$user"
```

Comment:
I like that this function `is_existing_user` attempts to do just that, but it is an expected side effect if it also creates as user as a result of performing the check. I prefer not to mix the concerns of checking and taking a specific action. Please move this elsewhere in the flow.